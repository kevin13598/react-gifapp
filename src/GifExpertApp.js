import React, {useState} from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ( { defaultCategories = [] } ) => {

    const [categories, setCategories] = useState(defaultCategories);

   /*  const handleAdd = (e) => {
        setCategories( categories => [...categories, 'HunterXHunter']);
        setCategories([...categories, 'HunterXHunter']);
    }; */

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories = {setCategories} />
            <hr></hr>
            { 
                    categories.map(category => 
                        <GifGrid 
                            key = {category}
                            category = {category} />
                    )
            }
                
        </>
    )
}