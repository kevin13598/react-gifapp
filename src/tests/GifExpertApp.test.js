import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import {GifExpertApp} from '../GifExpertApp';

describe('Pruebas en <GifExpertApp/>', () => {

    test('debe de mostrar el componente correctamente', () => {
        
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();

    })

    test('debe de mostrar una lista de categorias', () => {

        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories = {categories} />);

        expect(wrapper.find('GifGrid').length).toBe(categories.length);
        
    })
    
    

});