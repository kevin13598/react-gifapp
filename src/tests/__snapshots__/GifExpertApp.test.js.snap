// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp/> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
